<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Testing TRIAGE &#8212; TRIAGE v2 2.0.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=5ecbeea2" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=b08954a9" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=51b770b3"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Case Study" href="Case%20Study.html" />
    <link rel="prev" title="User-friendly Functions" href="User-friendly%20functions.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="testing-triage">
<h1>Testing TRIAGE<a class="headerlink" href="#testing-triage" title="Link to this heading">¶</a></h1>
<p>The TRIAGE R package (v2) offers a comprehensive suite of tools for analyzing transcriptomic data. This document provides a guide to testing the key functionalities of TRIAGE, including <cite>TRIAGEgene</cite>, <cite>TRIAGEcluster</cite>, and <cite>TRIAGEparser</cite>, along with their associated visualization and analysis functions. These tests are designed to demonstrate the capabilities of each function and ensure their correct operation.</p>
<section id="test-triagegene-plotjaccard-comparego">
<h2>Test TRIAGEgene + plotJaccard() + compareGO()<a class="headerlink" href="#test-triagegene-plotjaccard-comparego" title="Link to this heading">¶</a></h2>
<p><cite>TRIAGEgene</cite> is used for gene-level analysis and generating TRIAGE-weighted gene expression data.</p>
<p><strong># Test 1: Run TRIAGEgene on Demo Human Data</strong></p>
<p>Objective: To test <cite>TRIAGEgene</cite> using human data and generate visualizations, including a Jaccard Index heatmap and a dot plot for GO enrichment analysis.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Read the input file (tab delimited .txt file).</p></li>
<li><p>Run <cite>TRIAGEgene</cite> (Auto-selection for log transformation is enabled).</p></li>
<li><p>Generate a Jaccard Index Heatmap based on <cite>TRIAGEgene</cite> output (using top 100 genes by default).</p></li>
<li><p>Generate a dot plot to compare GO enrichment analysis based on <cite>TRIAGEgene</cite> output.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="c1"># Read input file</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEgene_demo_Human.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="n">demo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>

<span class="c1"># Run TRIAGEgene</span>
<span class="n">ds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">TRIAGEgene</span><span class="p">(</span><span class="n">demo</span><span class="p">)</span>

<span class="c1"># Generate a Jaccard Index Heatmap</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;/path/to/working/directory&quot;</span><span class="p">)</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;tests&quot;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;tests&quot;</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">plotJaccard</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tests/Jaccard_heatmap_Human_test1.pdf&quot;</span><span class="p">)</span>

<span class="c1"># Generate a dot plot to compare GO enrichment analysis based on DS results from &quot;group_1&quot;</span>
<span class="n">go_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">compareGO</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="n">ds_column</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group_1&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output_file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/group1_compareGO.pdf&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><strong># Test 2: Run TRIAGEgene on Demo Mouse Data</strong></p>
<p>Objective: To test <cite>TRIAGEgene</cite> using mouse data and generate visualizations.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Read the input file (CSV format).</p></li>
<li><p>Run <cite>TRIAGEgene</cite> with species specified as “Mouse”, with ‘pvalue’ enabled to calculate p-values using a rank-based Z-Score method. Auto-selection for log transformation is enabled.</p></li>
<li><p>Generate a Jaccard Index Heatmap using the top 100 genes.</p></li>
<li><p>Generate a dot plot to compare GO enrichment analysis based on <cite>TRIAGEgene</cite> output.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="c1"># Read input file (CSV)</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEgene_demo_Mouse.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="n">demo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>

<span class="c1"># Run TRIAGEgene for Mouse data</span>
<span class="n">ds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">TRIAGEgene</span><span class="p">(</span><span class="n">demo</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mouse&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">pvalue</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="c1"># Generate a Jaccard Index Heatmap</span>
<span class="nf">plotJaccard</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tests/Jaccard_heatmap_Mouse_test2.pdf&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">top_no</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>

<span class="c1"># Generate a dot plot to compare GO enrichment analysis based on DS results from &quot;cell_1&quot;, using top 20, 50, and 80 genes.</span>
<span class="n">go_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">compareGO</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="n">ds_column</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;cell_1&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">top_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="m">80</span><span class="p">),</span><span class="w"> </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;org.Mm.eg.db&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output_file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/cell1_compareGO.pdf&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><strong># Test 3: Run TRIAGEgene on Mouse Data with Matrix Input</strong></p>
<p>Objective: To evaluate the functionality of <cite>TRIAGEgene</cite> using mouse data in matrix format and generate a Jaccard Index Heatmap for visualization.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Read the input file and convert it to a matrix (CSV format).</p></li>
<li><p>Run <cite>TRIAGEgene</cite> with matrix input, specifying “Mouse” as the species.</p></li>
<li><p>Generate a Jaccard Index Heatmap using the top 88 genes.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="c1"># 1) Read input file (CSV) and convert to matrix</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEgene_demo_Mouse.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="n">demo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<span class="n">demo_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">demo</span><span class="p">)</span>

<span class="c1"># 2) Run TRIAGEgene with matrix input for Mouse data</span>
<span class="n">ds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">TRIAGEgene</span><span class="p">(</span><span class="n">demo_matrix</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mouse&quot;</span><span class="p">)</span>

<span class="c1"># 3) Generate Jaccard Index Heatmap</span>
<span class="nf">plotJaccard</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tests/Jaccard_heatmap_Mouse_test3.pdf&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">top_no</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">88</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="test-triagecluster-bypeak-topgenes">
<h2>Test TRIAGEcluster + byPeak() + topGenes()<a class="headerlink" href="#test-triagecluster-bypeak-topgenes" title="Link to this heading">¶</a></h2>
<p><cite>TRIAGEcluster</cite> is used for refining cell clustering in scRNA-seq data.</p>
<p><strong># Test 4: Run TRIAGEcluster and TRIAGEgene on Human Data</strong></p>
<p>Objective: To use <cite>TRIAGEcluster</cite> for cell clustering, <cite>byPeak()</cite> for analyzing average expression data by peak, and <cite>TRIAGEgene</cite> for generating TRIAGE-weighted expression data (DS).</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Run <cite>TRIAGEcluster</cite> for cell clustering, using CSV files for expression data and metadata, and select a suitable bandwidth based on UMAP reviews.</p></li>
<li><p>Run <cite>byPeak()</cite> to calculate average gene expression by peak.</p></li>
<li><p>Run <cite>TRIAGEgene</cite> to generate TRIAGE-weighted expression data.</p></li>
<li><p>Run <cite>topGenes()</cite> to extract the top 10 DS genes for each TRIAGE peak.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reticulate</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;/path/to/working/directory&quot;</span><span class="p">)</span>

<span class="c1"># 1) Run TRIAGEcluster</span>
<span class="n">expr_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEcluster_demo_expr_human.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="n">metadata_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEcluster_demo_metadata_human.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="nf">TRIAGEcluster</span><span class="p">(</span><span class="n">expr_file</span><span class="p">,</span><span class="w"> </span><span class="n">metadata_file</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/test4&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output_prefix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;demo&quot;</span><span class="p">)</span>

<span class="c1"># 2) Select a suitable bandwidth and run &#39;byPeak()&#39; to calculate average gene expression</span>
<span class="n">peak_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;tests/test4/demo_bw0.80_metadata.csv&quot;</span>
<span class="n">avg_peak</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">byPeak</span><span class="p">(</span><span class="n">expr_file</span><span class="p">,</span><span class="w"> </span><span class="n">peak_file</span><span class="p">,</span><span class="w"> </span><span class="n">cell_column</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Barcode&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">peak_column</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Peak&quot;</span><span class="p">)</span>
<span class="c1"># Save the average gene expression result to a CSV file</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="n">avg_peak</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/test4/AverageByPeak.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="c1"># 3) Run TRIAGEgene to generate TRIAGE-weighted expression data (DS)</span>
<span class="n">ds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">TRIAGEgene</span><span class="p">(</span><span class="n">avg_peak</span><span class="p">)</span>
<span class="c1"># Save the average DS result to a CSV file</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/test4/AverageByPeak_DS.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="c1"># Save the average DS result to a tab-delimited text file</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tests/test4/AverageByPeak_DS.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="c1"># 4) Run &#39;topGenes()&#39; to extract the top 10 genes for each TRIAGE peak based on DS values.</span>
<span class="n">top_ds_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topGenes</span><span class="p">(</span><span class="n">ds</span><span class="p">,</span><span class="w"> </span><span class="n">top_no</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="test-triageparser-plotgo-getclustergenes">
<h2>Test TRIAGEparser + plotGO() + getClusterGenes()<a class="headerlink" href="#test-triageparser-plotgo-getclustergenes" title="Link to this heading">¶</a></h2>
<p><cite>TRIAGEparser</cite> is a machine learning-based method for classifying genes with distinct biological functions.</p>
<p><strong># Test 5: Run TRIAGEparser with “AverageByPeak_DS.csv”</strong></p>
<p>Objective: To demonstrate <cite>TRIAGEparser</cite> functionality using a CSV file with four peak clusters.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Run <cite>TRIAGEparser</cite>.</p></li>
<li><p>Generate GO Heatmaps for All Groups.</p></li>
<li><p>Extract genes for cluster1 from the “Peak0_gene_clusters.csv” output of TRIAGEparser.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reticulate</span><span class="p">)</span>
<span class="c1"># 1) Run TRIAGEparser with &quot;AverageByPeak_DS.csv&quot; generated in Test 4</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;tests/test4/AverageByPeak_DS.csv&quot;</span>
<span class="nf">TRIAGEparser</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">input_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;table&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test5&quot;</span><span class="p">)</span>

<span class="c1"># 2) Generate Heatmaps using plotGO()</span>
<span class="nf">plotGO</span><span class="p">(</span><span class="n">indir</span><span class="o">=</span><span class="s">&quot;tests/test5&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test5&quot;</span><span class="p">)</span>

<span class="c1"># 3) Extract genes for cluster1 from the &quot;Peak0_gene_clusters.csv&quot; using getClusterGenes()</span>
<span class="n">cluster1_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getClusterGenes</span><span class="p">(</span><span class="s">&quot;tests/test5/gene_clusters/Peak0_gene_clusters.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cluster1&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><strong># Test 6: Run TRIAGEparser with “AverageByPeak_DS.txt”</strong></p>
<p>Objective: To demonstrate <cite>TRIAGEparser</cite> functionality using a tab-delimited text file and generate a specific gene group heatmap.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Run <cite>TRIAGEparser</cite> with tab-delimited text file input.</p></li>
<li><p>Generate GO Heatmap for the “Peak0” group.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">TRIAGE</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reticulate</span><span class="p">)</span>
<span class="c1"># 1) Run TRIAGEparser with &quot;AverageByPeak_DS.txt&quot; generated in Test 4</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;tests/test4/AverageByPeak_DS.txt&quot;</span>
<span class="nf">TRIAGEparser</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">input_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;table&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test6&quot;</span><span class="p">)</span>

<span class="c1"># 2) Generate heatmap for &quot;Peak0&quot; group</span>
<span class="nf">plotGO</span><span class="p">(</span><span class="n">indir</span><span class="o">=</span><span class="s">&quot;tests/test6&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test6&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Peak0&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><strong># Test 7: Run TRIAGEparser with a Gene List</strong></p>
<p>Objective: To test <cite>TRIAGEparser</cite> using a gene list and visualize gene ontology enrichment.</p>
<p><strong>Steps:</strong></p>
<ol class="arabic simple">
<li><p>Run <cite>TRIAGEparser</cite> with a gene list file as input.</p></li>
<li><p>Generate Gene Ontology Heatmap.</p></li>
</ol>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1"># 1) Run TRIAGEparser with gene list file</span>
<span class="n">input_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TRIAGEparser_demo_genelist.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TRIAGE&quot;</span><span class="p">)</span>
<span class="nf">TRIAGEparser</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span><span class="w"> </span><span class="n">input_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;list&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test7&quot;</span><span class="p">)</span>

<span class="c1"># 2) Generate Gene Ontology Heatmap</span>
<span class="nf">plotGO</span><span class="p">(</span><span class="n">indir</span><span class="o">=</span><span class="s">&quot;tests/test7&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">outdir</span><span class="o">=</span><span class="s">&quot;tests/test7&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>These tests serve as a practical demonstration of how to apply the TRIAGE R package for analyzing and visualizing complex transcriptomic data. Researchers can adapt these procedures to their specific datasets, ensuring the effective use of TRIAGE in research projects.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">TRIAGE v2</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="TRIAGEgene.html">TRIAGEgene</a></li>
<li class="toctree-l1"><a class="reference internal" href="TRIAGEcluster.html">TRIAGEcluster</a></li>
<li class="toctree-l1"><a class="reference internal" href="TRIAGEparser.html">TRIAGEparser</a></li>
<li class="toctree-l1"><a class="reference internal" href="User-friendly%20functions.html">User-friendly Functions</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Testing TRIAGE</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#test-triagegene-plotjaccard-comparego">Test TRIAGEgene + plotJaccard() + compareGO()</a></li>
<li class="toctree-l2"><a class="reference internal" href="#test-triagecluster-bypeak-topgenes">Test TRIAGEcluster + byPeak() + topGenes()</a></li>
<li class="toctree-l2"><a class="reference internal" href="#test-triageparser-plotgo-getclustergenes">Test TRIAGEparser + plotGO() + getClusterGenes()</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Case%20Study.html">Case Study</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="User-friendly%20functions.html" title="previous chapter">User-friendly Functions</a></li>
      <li>Next: <a href="Case%20Study.html" title="next chapter">Case Study</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Qiongyi Zhao.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.2.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/Testing TRIAGE.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>